src/main/java/io/specto/hoverfly/junit/core/model/FieldMatcherDeserializer.java

```diff
@@ -0,0 +1,26 @@
+package io.specto.hoverfly.junit.core.model;
+
+import com.fasterxml.jackson.core.*;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+
+class FieldMatcherDeserializer extends JsonDeserializer<FieldMatcher> {
+    @Override
+    public FieldMatcher deserialize(JsonParser jsonParser, DeserializationContext context) throws IOException {
+
+        FieldMatcher matcher = null;
```

Comment:
As in, how is there a situation where it can be an object or text. Could you show me an example, please?